resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -661,6 +665,33 @@ private[spark] class Client(
       }).toMap
     }).getOrElse(Map.empty[String, String])
   }
+
+  private def checkForUserJarsFilesExistence: Unit = {
+    val conf = new Configuration
+
+    def existsOrAbort(uri: URI): Unit = {
+      val path = new Path(uri)
+      if (!path.getFileSystem(conf).isFile(path)) {
+        throw new SparkException(s"""file "${uri}" does not exist!""")
+      }
+    }
+
+    def checkExistence(maybePaths: Option[String]): Unit = {
```

Comment:
Again, avoid nested methods. We can make this be the outermost method, and then `run()` can just call this three times, once for each of the appropriate fields to inspect.